### PR TITLE
Develop csv import

### PR DESCRIPTION
**Pull request by GiaJordan**

### CODE REVIEW

File: schematic/models/metadata.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 # we shouldn't need to expose Synapse functionality explicitly
 from schematic.store.synapse import SynapseStorage
 
-from schematic.utils.df_utils import trim_commas_df
+from schematic.utils.df_utils import trim_commas_df, load_df
```
Comment (milen-sage) [inline schematic/models/metadata.py:25]: Do we still need trim_commas_df?

**Review by milen-sage — APPROVED** [review]
@GiaJordan this looks great! Just one minor  question in code

File: schematic/models/metadata.py (hunk 1 above)
Comment (GiaJordan) [reply schematic/models/metadata.py:25]: I think we'll be able to do without it now!

**Comment by GiaJordan** [comment]
@allaway As of the merge of this PR the CSV import for schematic is standardized. To make the errors generated from the validation suite more meaningful, changes were made to the datatypes of columns and cells of imported manifest CSVs. 

Previously, each cell was imported as string type but now numerical cells will be the appropriate type: either an `int` or `float` type instead of `string`. All non-numerical text will still be type `string`. I know that previously you updated the [csv read function](https://github.com/Sage-Bionetworks/schematic/commit/e7ce4ba58f0a38af4a02fa65e13f21a4dd75ea07) to import everything as type string, so I wanted to make you aware of this change.

Can you let @milen-sage and I know how this new import method works with your workflow, and if you're able to adapt your workflow to work with the new standard should there be disparities?

**Comment by allaway** [comment]
Hi @GiaJordan!

The main reason I had implemented that change back in #539 was because `schematic` didn't have any more sophisticated logic at the time to ensure that the data was preserved. I'm not too worried about preserving the type as string - it was just a lazy fix to ensure that the data itself wasn't changed to something different (e.g. TRUE->True, 1->1.0) etc. 

It sounds like this PR will fix both my issues #664 and #600. I can pull these changes into our staging DCA and close them if so!

**Comment by GiaJordan** [comment]
It should address #664 but from what I've gathered it think there's also something happening before schematic when the manifest is created that influences the number value read. The manifests that I've created and uploaded during my testing process for this did not have the issue you described, but this fix was intended for something slightly different so there may be more at play for that specifically.

I'm not sure about #600, so I'll likely need to investigate a fix for that specifically at a later date because it seems like those values have been read as strings for a while and still are here.

We'll be merging more associated changes within the next day with PR #682 that includes some more logic for data model CSVs and manifests when they're not being read for validation purposes.
If you'd like to test it though you can pull this branch now! Otherwise I can put those issues on my backlog as appropriate.
@allaway

**Comment by allaway** [comment]
Hey @GiaJordan - thanks for the comments, that's really helpful.
I updated the DCA including the #682 PR. I am running into an issue where integers with *defined* values are being flagged as invalid (probably because the defined values in the schema are treated as strings, but `schematic`is reading in the CSV and interpreting these as integers). Do you think this is an issue on my end (maybe we need to define this in the schema differently to have schematic treat the schema values as integers?) or is this a bug in how schematic is interpreting the schema values? Unfortunately we can't simply set`validation` to be integers, because there are a limited number of valid values, it's not all integers. 

Screenshot example: 
Screen Shot 2022-04-27 at 7 51 15 PM

**Comment by allaway** [comment]
If this seems like a `schematic` bug and not just something we need to fix in the schema, I'll file this as a proper issue :) let me know what you think!

**Comment by GiaJordan** [comment]
@allaway addressed in #686!
**cc** @milen-sage

**Pull request merged**